dts/common/yaml/i2c.yaml

```diff
@@ -0,0 +1,25 @@
+---
+title: I2C Base Structure
+id: i2c
+version: 0.1
+
+description: >
+    This binding gives the base structures for all I2C devices
+
+properties:
+  - clock-frequency :
+      type: int
+      category: optional
+      description: Desired I2C bus clock frequency in Hz
```

Comment:
According to the I2C standard: "The bus speed is always constrained to the maximum communication rate of the slowest device attached to the bus". So this is the one important piece of information which needs to be provided by DTS. Otherwise I2C driver has no way of knowing what is the maximum speed it is allowed to run the bus at.

Maybe we should change the description to "Maximum I2C bus clock frequency in Hz"? Or do we need a second property?